src/components/EnterpriseList/index.jsx

```diff
@@ -126,23 +173,33 @@ class EnterpriseList extends React.Component {
 
   render() {
     const { loading, error } = this.props;
-    const { enterprises, pageCount } = this.state;
-
+    const { enterprises, searchQuery } = this.state;
     return (
       <div>
         <Helmet>
           <title>Enterprise List</title>
         </Helmet>
         <div className="container">
           <div className="row mt-4">
-            <div className="col">
+            <div className="col-sm-12 col-md">
               <H1>Enterprise List</H1>
+            </div>
+            <div className="col-sm-12 col-md-4 col-lg-3 mb-3 mb-md-0">
+              <SearchBar
+                onSearch={query => this.handleSearch(query)}
+                onClear={() => this.handleClear()}
+                value={searchQuery}
+              />
+            </div>
+          </div>
+          <div className="row mt-2">
+            <div className="col">
               {error && this.renderErrorMessage()}
               {loading && !enterprises && this.renderLoadingMessage()}
               {!loading && !error && enterprises && enterprises.length === 0 &&
                 this.renderEmptyEnterpriseListMessage()
               }
-              {!loading && !error && enterprises && enterprises.length === 1 && pageCount === 1 &&
+              {this.shouldRenderRedirectToEnterpriseAdminPage() &&
```

Comment:
Moving this conditional to its own function to reduce some complexity in the `render` function.